### ISSUE TITLE
migrating from mio-extras

### ISSUE THREAD

**Issue by danieleades** [onurzdg/mio-misc#8]
hi there, i'm looking at migrating a crate from mio 0.6 -> 0.7. This also necessitates (I believe) migrating from mio-extras to mio-misc.

The crate uses `mio_extras::channel::Receiver`, which implements mio 0.6's `mio::Evented`. The equivalent trait in mio 0.7, `mio::event::Source` is *not* implemented for the `Receiver` type in `mio-misc` which is a re-export of `std::sync::mspc::Receiver`.

how should these be reconciled?

for reference, the crate is here - https://github.com/jhelovuo/RustDDS

**Comment by onurzdg** [comment]
I will try to get to this in a couple of weeks.  In the meantime, I am also OK with looking at a PR that addresses the issue.

**Comment by onurzdg** [comment]
@danieleades Sorry to disappoint you, but I don't think what you are asking for is possible anymore. `Evented` was implemented for Receiver in mio-extras to keep track of `readiness`.  Readiness abstraction is gone now: https://tokio.rs/blog/2019-12-mio-v0.7-alpha.1

There's the `Waker` abstraction now. I would encourage you to check out the [tests](https://github.com/onurzdg/mio-misc/blob/master/test/test_poll_channel.rs) I wrote for mio-misc, If you haven't already. You maybe able to satisfy your use case via NotificationQueue and Waker.
